tests/cypress/specs/rate-limit.cy.js

```diff
@@ -1,64 +1,62 @@
-const agencies = require("../fixtures/transit-agencies");
 const users = require("../fixtures/users.json");
+const helpers = require("./helpers");
 const { eligibility_url, post_confirm } = require("../plugins/eligibility");
 
+const SUCCESS_STATUS = 302;
+const FAIL_STATUS = 503;
+
 describe("Rate limiting feature spec", () => {
   beforeEach(() => {
     cy.visit("/");
 
-    // agency selection
-    cy.contains("Choose Your Provider").click();
-    cy.contains(agencies[0].fields.long_name).click();
-
-    // select Courtesy Card
-    // TODO find a more robust way to do this
-    cy.get('#form-verifier-selection [type="radio"]').check("2");
-    cy.get("#form-verifier-selection button[type='submit']").click();
-    cy.contains("Continue").click();
+    helpers.selectAgency();
+    helpers.selectCourtesyCard();
   });
 
   it("Limits excess requests", () => {
     const sub = users.ineligible.sub;
     const name = users.ineligible.name;
 
-    // start by making 5 times as many requests as allowed
-    // these should match the defaults in settings.py
-    const RATE_LIMIT = 5;
-    const N_REQUESTS = RATE_LIMIT * 5;
-    const RATE_LIMIT_PERIOD = 60;
-
-    const SUCCESS_STATUS = 302;
-
     // csrfmiddlewaretoken value needed to enable automated form submissions
     // there could be multiple forms on the page, so get() the one with the
     // action we are interested in
     cy.get(`form[action='${eligibility_url}'`)
       .find("input[name='csrfmiddlewaretoken']")
       .invoke("attr", "value")
       .then((csrfmiddlewaretoken) => {
-        // make excess requests
-        for (let i = 0; i < N_REQUESTS; i++) {
-          // continue on non-successful status codes to check response
+        // make successive requests with no wait time
+        for (let i = 0; i < helpers.RATE_LIMIT; i++) {
           post_confirm(csrfmiddlewaretoken, sub, name, false).then((res) => {
-            if (i < RATE_LIMIT) {
-              // allow up to API_RATE_LIMIT requests
+            // the first request should succeed, subsequent should fail
+            if (i == 0) {
               expect(res.status).to.eq(SUCCESS_STATUS);
             } else {
-              // we've gone beyond API_RATE_LIMIT
-              expect(res.status).to.eq(400);
+              expect(res.status).to.eq(FAIL_STATUS);
```

Comment:
I think the nginx rate-limiting configuration / implementation is working, but these tests may have a bug.

I noticed the Cypress test log wasn't showing me any `expect` commands. Digging into it more, I see `helpers.RATE_LIMIT` is undefined, so the loop is a no-op.

![image](https://user-images.githubusercontent.com/25497886/232881784-0dac5e11-742f-4ab4-a11a-e75df7b93712.png)

Added a log statement:
![image](https://user-images.githubusercontent.com/25497886/232881960-24270f99-446e-43e5-9086-2e9df752a158.png)

![image](https://user-images.githubusercontent.com/25497886/232882032-a3f68b36-c85e-445f-88d0-2e9868219de8.png)

Reply:
Definitely seems like a bug. Not sure, I had this working last week, it was definitely running the rate limit test (took multiple minutes with the `cy.wait()` -- but now it's finishing in under 10 seconds so not running!

Reply:
OK this was my fault, I had it working when those `RATE_LIMIT` consts where defined in the same spec file. I forgot to `export` them from the `helpers.js` file though, hence not being available.

Fix incoming.

Reply:
Do you have any ideas on why my test is failing to get a 302 for the first request? Your screenshot of the GitHub Action run shows that this should work...

![image](https://user-images.githubusercontent.com/25497886/232901704-aeb13146-1f5a-4528-b1b6-3d30b61b970a.png)

Reply:
To mirror the GitHub Action:

* Make sure `.env` is empty:

  ```
  ```
  
* Recreate both services:

```bash
docker compose down
docker compose up -d client server
```

* Run cypress tests _with the correct base URL_:

```bash
cd tests/cypress
CYPRESS_baseUrl=http://localhost:8000 npx cypress run
```

Reply:
Thanks! I got it to pass